src/Microsoft.Framework.ConfigurationModel.UserSecrets/PathHelper.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Framework.Internal;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Framework.ConfigurationModel.UserSecrets
+{
+    public class PathHelper
+    {
+        private const string SECRETS_FILE_NAME = "secrets.json";
+
+        public static string GetSecretsPath([NotNull]string projectPath)
+        {
+            var projectFilePath = Path.Combine(projectPath, "project.json");
+
+            if (!File.Exists(projectFilePath))
+            {
+                throw new InvalidOperationException(
+                    string.Format(Resources.Error_Missing_Project_Json, projectFilePath));
+            }
+
+            var obj = JObject.Parse(File.ReadAllText(projectFilePath));
+            var userSecretsId = obj.Value<string>("userSecretsId");
+
+            if (string.IsNullOrEmpty(userSecretsId))
+            {
+                throw new InvalidOperationException(
+                    string.Format(Resources.Error_Missing_UserSecretId_In_Project_Json, projectFilePath));
+            }
+
+            var badCharIndex = userSecretsId.IndexOfAny(Path.GetInvalidPathChars());
+            if (badCharIndex != -1)
+            {
+                throw new InvalidOperationException(
+                    string.Format(
+                        Resources.Error_Invalid_Character_In_UserSecrets_Id,
+                        userSecretsId[badCharIndex],
+                        badCharIndex));
+            }
+
+            var root = Environment.GetEnvironmentVariable("APPDATA") ??         // On Windows it goes to %APPDATA%\Microsoft\UserSecrets\
+                        Environment.GetEnvironmentVariable("HOME");             // On Mac/Linux it goes to ~/Microsoft/UserSecrets/
+
+            return Path.Combine(root, "Microsoft", "UserSecrets", userSecretsId, SECRETS_FILE_NAME);
```

Comment:
Hmm. To keep it simple I'll have to assume the current OS is `Windows` if environment variable `APPDATA` is available and assume it is non-windows on its absense. We need a better way to figure out the OS. I'll file a work item in Configuration repo to update it when we have the `OSVersion` available across.